Add 'hostPoolSize' option to ConnectionPool

hpainter opened the pull request:
Small modification that allows the number connections per cluster host to be increased.  Similar to the node-postgres module, which uses a pool of 10 'parallel' connections to the db, bumpng this number up increased the throughput of our app which was making a lot of asynchronous read requests to the cassandra db.

Commit:
Add 'hostPoolSize' option to ConnectionPool.

Allows the number of connections per host to be modified.  Defaults
to 1 if not specified.

devdazed commented:
Awesome, commit. I've been meaning to do this [ ref #21 ].  The only thing I ask before I pull is that a test get written for it.  I want to ensure we don't break existing functionality.

Commit:
Move counter var initialization.

Fixes a bug where doing the counter initialization up top prevented
connections to hosts after the first one.

hpainter commented:
I'll see if I can get some test coverage for this.  I noticed that if I set multiple hosts in tests/helpers/connection.js it causes some tests to fail, and I suspect I can get a fix for that in with this as well.

Commit:
Move onConnect callback from first to final connection.

In ConnectionPool.connect(), if the callback gets fired after making the
first successfull connection then subsequent ConnectionPool.use() calls may
not set the keyspace on the entire set of connections.  This moves the
callback to happen after the final connection is set.

Commit:
Test suite for ConnectionPools that use hostPoolSize option.

Re-runs the existing thrift.js test suite, but sets the
hostPoolSize option before starting.

hpainter commented:
OK, tests are ready and passing.  The current suite of thrift.js tests just get imported and re-run with the hostPoolSize option set.  Let me know if you see any issues...
